src/test/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFModelUnitTest.java

```diff
@@ -0,0 +1,246 @@
+package org.broadinstitute.hellbender.tools.exome.allelefraction;
+
+import htsjdk.samtools.util.Interval;
+import org.apache.commons.math3.special.Gamma;
+import org.broadinstitute.hellbender.tools.exome.AllelicCount;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.Math.log;
+
+/**
+ * Created by davidben on 11/24/15.
+ */
+public class AFModelUnitTest {
+    final static Interval DUMMY = new Interval("dummy", 1, 2);
+    final double EPSILON = 1e-10;
+    /**
+     * We subject the likelihood (with bias marginalized out) to several tests.  Recall that the exact likelihood is:
+     *
+     * log { [beta^alpha / Gamma(alpha)][(1-pi)/2] * int_{0 to infty} f^a * (1-f)^r * lambda^(alpha + r - 1) * exp(-beta*lambda)/(f + (1-f)*lambda)^n d lambda }
+     * if indicator == ALT_MINOR,
+     *
+     * same as ALT_MINOR case but with f <--> 1 - f
+     * if indicator == REF_MINOR,
+     *
+     * log {pi * a!r!/(n+1)!}
+     * if indicator == OUTLIER,
+     *
+     * where alpha = mu*beta and n = a + r
+     * */
+
+    //first test f = EPSILON where EPSILON << 1: we have an analytic result for comparison
```

Comment:
The tests all use different parameters so I think this would just result in moving the nested loops to the `DataProviders`.